main.py

```diff
@@ -207,21 +207,25 @@ def plot_data(new, old):
     input_fifo = os.open(input_path, os.O_RDONLY)
     output_fifo = os.open(output_path, os.O_WRONLY)
 
-    while True:
-        if args.verbose:
-            print("reading")
+    if args.verbose:
+        print("reading")
+
+    binary_content_data = os.read(input_fifo, DATA_LENGTH)
+    if len(binary_content_data) != DATA_LENGTH:
+        exit(1)
```

Comment:
maybe in another PR, but we've been using [syslog](https://docs.python.org/3/library/syslog.html) for debugging this kind of thing.

anywhere you have a `print` consider instead using `syslog` if it's useful runtime debug info.

Reply:
Ok ill add that in another PR

Reply:
Should those be underneath a verbose flag?

Reply:
i'll leave it up to you -- we can start the program with verbose flags set, or it could be default.

we've been `syslog`ing things when
- they're fatal errors
- thing is working correctly, but that is infrequent and 'major' (don't want to spam logs)